### PR TITLE
Migrate data trie

### PR DESCRIPTION
**Pull request by BeniaminDrasovean**

### CODE REVIEW

File: core/interface.go [hunk 1]
```diff
@@ -136,3 +135,15 @@ type GetNodeFromDbErrHandler interface {
...
+
+// TrieNodeVersionVerifier defines the behavior of a component that can verify if a trie node version is valid
+type TrieNodeVersionVerifier interface {
+	IsValidVersion(version TrieNodeVersion) bool
+	IsInterfaceNil() bool
+}
+
+// EnableEpochsHandler defines the behavior of a component that can return if a feature is enabled or not
+type EnableEpochsHandler interface {
+	IsAutoBalanceDataTriesEnabled() bool
```
Comment (iulianpascalau) [inline core/interface.go:147]: :+1:

File: core/trie.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+const (
+	// NotSpecified means that the value is not populated or is not important
+	NotSpecified TrieNodeVersion = iota
+
+	// AutoBalanceEnabled is used for data tries, and only after the activation of AutoBalanceDataTriesEnableEpoch flag
+	AutoBalanceEnabled
+)
+
+// GetStringForVersion returns the string representation of the given trie node version
+func GetStringForVersion(version TrieNodeVersion) string {
```
Comment (iulianpascalau) [inline core/trie.go:21]: For easier integration, can we move the code of this function on the TrieNodeVersion and call the new function `String() string` ?

File: core/trie.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+// IsValidVersion returns true if the given trie node version is valid
+func (vv *trieNodeVersionVerifier) IsValidVersion(version TrieNodeVersion) bool {
+	if vv.enableEpochsHandler.IsAutoBalanceDataTriesEnabled() {
+		return version <= AutoBalanceEnabled
+	}
+
+	return version == NotSpecified
+}
+
+func (vv *trieNodeVersionVerifier) IsInterfaceNil() bool {
```
Comment (iulianpascalau) [inline core/trie.go:55]: missing comment

File: core/trie.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+	AutoBalanceEnabled
+)
+
+// GetStringForVersion returns the string representation of the given trie node version
+func GetStringForVersion(version TrieNodeVersion) string {
+	switch version {
+	case NotSpecified:
+		return "not specified"
+	case AutoBalanceEnabled:
+		return "auto balanced"
```
Comment (ssd04) [inline core/trie.go:26]: maybe constants for these?

Comment (BeniaminDrasovean) [reply core/trie.go:26]: Done.

**Review by iulianpascalau — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**